orttraining/orttraining/training_ops/cuda/nn/dropout.cc

```diff
@@ -102,5 +102,112 @@ Status DropoutGrad<T1, T2>::ComputeInternal(OpKernelContext* context) const {
 
   return Status::OK();
 }
+
+#define REGISTER_BIAS_DROPOUT_KERNEL_TYPED(T1, T2)                   \
+  ONNX_OPERATOR_TYPED_KERNEL_EX(                                     \
+      BiasDropout,                                                   \
+      kMSDomain,                                                     \
+      1,                                                             \
+      T1##_##T2,                                                     \
+      kCudaExecutionProvider,                                        \
+      KernelDefBuilder()                                             \
+          .TypeConstraint("T", DataTypeImpl::GetTensorType<T1>())    \
+          .TypeConstraint("T1", DataTypeImpl::GetTensorType<T2>())   \
+          .TypeConstraint("T2", DataTypeImpl::GetTensorType<bool>()) \
+          .InputMemoryType<OrtMemTypeCPUInput>(3)                    \
+          .InputMemoryType<OrtMemTypeCPUInput>(4),                   \
+      BiasDropout<T1, T2>);
+
+REGISTER_BIAS_DROPOUT_KERNEL_TYPED(MLFloat16, MLFloat16)
+REGISTER_BIAS_DROPOUT_KERNEL_TYPED(MLFloat16, float)
+REGISTER_BIAS_DROPOUT_KERNEL_TYPED(float, MLFloat16)
+REGISTER_BIAS_DROPOUT_KERNEL_TYPED(float, float)
+
+template <typename T1, typename T2>
+Status BiasDropout<T1, T2>::ComputeInternal(OpKernelContext* context) const {
+  typedef typename ToCudaType<T1>::MappedType CudaT;
+ 
+  //Get X_data
+  const Tensor* X = context->Input<Tensor>(0);
+  if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "X Input is not available.");
```

Comment:
>if (X == nullptr) return Status(common::ONNXRUNTIME, common::FAIL, "X Input is not available."); [](start = 2, length = 96)

`if (!cond) return Status(...)` -> `ORT_RETURN_IF_NOT(cond, "...")` to be more concise